src/main/java/focusedCrawler/crawler/async/FetchedResultHandler.java

```diff
@@ -38,11 +48,30 @@ public void failed(LinkRelevance link, Exception e) {
         if(e instanceof AbortedFetchException) {
             AbortedFetchException afe = (AbortedFetchException) e;
             logger.info("Download aborted: \n>URL: {}\n>Reason: {}", link.getURL().toString(), afe.getAbortReason());
-        } else {
+        }else if(e.getCause() instanceof SSLException){
+            if(!sslExceptions.contains(link.toString())){
+                handleSSLExceptions(link);
+            }else{
+                logger.info("Failed to download URL: {}\n>Tried both HTTP and HTTPS. Reason: {}", link.getURL().toString(), e.getMessage());
+            }
+        }else {
             logger.info("Failed to download URL: {}\n>Reason: {}", link.getURL().toString(), e.getMessage());
         }
     }
-    
+    private void handleSSLExceptions(LinkRelevance link){
```

Comment:
Instead of inserting a new URL in the link storage, fixing the Fetcher to work with HTTPS redirects would be a more robust solution.

Can you invest more time in finding a solution that involves changing only the fetcher, i.e, making sure that the fetcher is be able to follow redirects to https links automatically without involving the other parts of the crawler?

That may involve configuring correctly the Apache httpclient library inside the SimpleHttpFetcher class.